src/messages.rs

```diff
@@ -583,6 +632,22 @@ impl Debug for DirectMessage {
             DirectMessage::TunnelDisconnect(peer_id) => {
                 write!(formatter, "TunnelDisconnect({:?})", peer_id)
             }
+            DirectMessage::ResourceProof { ref seed, ref target_size, ref difficulty } => {
+                write!(formatter,
+                       "ResourceProof {{ {:?}, {:?}, {:?} }}",
```

Comment:
I think we should at least annotate the `target_size` and `difficulty` fields as they're both `u32`s.